app/controllers/foreman_tasks/api/tasks_controller.rb

```diff
@@ -245,6 +246,13 @@ def search_tasks(search_params)
         scope.all.map { |task| task_hash(task) }
       end
 
+      def current_taxonomy_search
+        conditions = []
+        conditions << "organization_id = #{Organization.current.id}" if Organization.current
+        conditions << "location_id = #{Location.current.id}" if Location.current
+        conditions.empty? ? '' : "(#{conditions.join(' AND ')})"
+      end
```

Comment:
We already have the exact same method in the non-api controller. Could we maybe unify the two?

Reply:
yup, I've extracted the method from there.
I've tried to unify, but keep the changes minimal.